lib/Platform/Intl/impl_icu/LocaleResolver.h

```diff
@@ -0,0 +1,159 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#ifndef HERMES_PLATFORMINTL_IMPLICU_LOCALERESOLVER_H
+#define HERMES_PLATFORMINTL_IMPLICU_LOCALERESOLVER_H
+
+#include "LocaleBCP47Object.h"
+#include "hermes/Platform/Intl/PlatformIntl.h"
+
+#include <map>
+#include <string>
+#include <string_view>
+#include <unordered_set>
+
+namespace hermes {
+namespace platform_intl {
+namespace impl_icu {
+
+class LocaleResolver {
```

Comment:
Does this need to be a class? It seems a namespace would suffice.

Reply:
Done in 3rd commit.